akka-cluster-typed/src/main/scala-3/akka/cluster/typed/internal/receptionist/ClusterReceptionistProtocol.scala

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (C) 2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.cluster.typed.internal.receptionist
+
+import akka.actor.typed.ActorRef
+import akka.actor.typed.internal.receptionist.{AbstractServiceKey, ReceptionistMessages}
+import akka.annotation.InternalApi
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[receptionist] object ClusterReceptionistProtocol {
+  type Aux[P] = AbstractServiceKey { type Protocol = P }
+
+  type SubscriptionsKV[K <: Aux[_]] = K match {
+    case Aux[t] => ActorRef[ReceptionistMessages.Listing[t]]
```

Comment:
Please, remind me again what is the trick with `t` (lowercase).

Reply:
in this case it's a variable in the type-level patternmatch, https://blog.rockthejvm.com/scala-3-match-types/